submission/submission-webapp/src/itest/kotlin/ac/uk/ebi/biostd/itest/test/submission/submit/SubmissionDraftApiTest.kt

```diff
@@ -243,4 +244,48 @@ class SubmissionDraftApiTest(
             assertThat(thirdVersion.modificationTime).isAfter(secondVersion.modificationTime)
             webClient.deleteSubmissionDraft(accNo)
         }
+
+    @Test
+    fun `12-9 submit json when a draft already exists`() =
+        runTest {
+            val accNo = "ABC-131"
+            val subV1 =
+                jsonObj {
+                    "accno" to accNo
+                    "section" to
+                        jsonObj {
+                            "type" to "Study"
+                        }
+                    "type" to "submission"
+                }.toString()
+
+            assertThat(webClient.submit(subV1)).isSuccessful()
+            webClient.getSubmissionDraft(accNo)
+            webClient.updateSubmissionDraft(
+                accNo,
+                jsonObj {
+                    "accno" to accNo
+                    "section" to
+                        jsonObj {
+                            "type" to "ByDraft"
+                        }
+                    "type" to "submission"
+                }.toString(),
+            )
+
+            val subV2 =
+                jsonObj {
+                    "accno" to accNo
+                    "section" to
+                        jsonObj {
+                            "type" to "Another"
+                        }
+                    "type" to "submission"
+                }.toString()
+            val response = webClient.submit(subV2)
+            assertThat(response).isSuccessful()
+
+            val updatedSubmission = response.body
+            assertThat(updatedSubmission.section.type).isEqualTo("Another")
```

Comment:
should something happened here with the draft? like should it be deleted?

Reply:
It should be marked as `PROCESSED`. The concept of deleting the draft after deletion no longer applies since now it's a single entity for the whole process. I didn't verify it because it's already checked in other tests.